actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -19,35 +19,36 @@ module ActionController
   #   params.require(:a)
   #   # => ActionController::ParameterMissing: param is missing or the value is empty: a
   class ParameterMissing < KeyError
-    attr_reader :param, :keys # :nodoc:
-
-    def initialize(param, keys = nil) # :nodoc:
-      @param = param
-      @keys  = keys
-      super("param is missing or the value is empty: #{param}")
-    end
-
-    class Correction
-      def initialize(error)
-        @error = error
+    # In Ruby <= 2.5 +KeyError#new+ does not take the +key+ or +receiver+
+    # arguments and we need this feature detection.
+    takes_key_and_receiver = begin
+                               !!new(nil, key: nil, receiver: nil)
+                             rescue ArgumentError
+                               false
+                             end
```

Comment:
@tenderlove I wonder how you feel about this... We could remove this feature detection if we could drop support for 2.5.

Reply:
I don’t know if we can drop 2.5, but let me ask the rest of the core team.  I’m personally OK with it since we only use 2.6 / 2.7, but I’m not sure only one data point is a good enough survey 😆